### PR TITLE
Adds cli

### LINKED ISSUE #1: CLI: Python project scaffolding (pyproject, dependencies, typer)

### PR DESCRIPTION
**Pull request by cpapad**
## 📝 Description

This Pr adds interactive message selection functionality with edit capabilities and a basic loading animation for regeneration action
#### Before

_{Please add a screenshot here}_

#### After

_{Please add a screenshot here}_

## ℹ️ Issue
#15

### CODE REVIEW

File: g3/cli/cli.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+
```
Comment (antonakospanos) [inline g3/cli/cli.py:1]: Shall we move this under `g3/generate/preview` package since it's going to be used for the preview of the generated messages?

**Review by antonakospanos — APPROVED** [review]

File: g3/cli/cli.py (hunk 1 above)
Comment (cpapad) [reply g3/cli/cli.py:1]: Yes of course, there is absolutely no strong opinion on this. Just placed them and named them to get going

**Review by ikarachristos — APPROVED** [review]
Cool addition. Inquirer also provides editor integration which will probably be useful as well.

**Comment by cpapad** [comment]
> Cool addition. Inquirer also provides editor integration which will probably be useful as well.

Initially we used the inquirer editor (which uses the same library we used) but had an extra step (press enter to enter editor) so we used directly the library to avoid that

**Pull request merged**